components/filesystem.rst

```diff
@@ -315,6 +331,191 @@ contents at the end of some file::
 If either the file or its containing directory doesn't exist, this method
 creates them before appending the contents.
 
+Path
+----
+
+.. versionadded:: 5.4
+
+    The :class:`Symfony\\Component\\Filesystem\\Path` class was introduced in Symfony 5.4.
+
+Dealing with file paths usually involves some difficulties:
+
+- System Heterogeneity: file paths look different on different platforms. UNIX
+  file paths start with a slash ("/"), while Windows file paths start with a
+  system drive ("C:"). UNIX uses forward slashes, while Windows uses backslashes
+  by default ("").
+- Absolute/relative paths: web applications frequently need to deal with absolute
+  and relative paths. Converting one to the other properly is tricky and
+  repetitive.
+
+:class:`Symfony\\Component\\Filesystem\\Path` provides utility methods to tackle
+those issues.
+
+Canonicalization
+~~~~~~~~~~~~~~~~
+
+*Canonicalization* is the transformation of a path into a normalized (the
```

Comment:
This description is a bit confusing to me. We could use something like the short explanation given for Golang `Path::clean()` (https://pkg.go.dev/path#Clean):

> Returns the shortest path name equivalent to the given path. It applies the following rules iteratively until no further processing can be done:
>
> (and list here the rules applied by Symfony)